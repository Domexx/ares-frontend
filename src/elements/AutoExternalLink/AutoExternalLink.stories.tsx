import { Story } from '@storybook/react/types-6-0';
import React, { ComponentProps } from 'react';
import AutoExternalLinkComponent, { Props } from './AutoExternalLink';

const Component = {
  title: 'Elements/Auto External Link',
  component: AutoExternalLinkComponent,
  argTypes: {
    isExternal: { control: { type: 'boolean' } },
  },
};

const Template: Story<ComponentProps<typeof AutoExternalLinkComponent>> = (args) => {
  return <AutoExternalLinkComponent {...args} />;
};

export const AutoExternalLink = Template.bind({});
export default Component;

AutoExternalLink.args = {
  url: 'https://virgil-abi.com',
  text: 'Jelly jelly virgil',
} as Props;
