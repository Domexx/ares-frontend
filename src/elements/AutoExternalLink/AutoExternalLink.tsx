import React from 'react';
import { isExternalLink } from '../../shared/helpers';
import Link from '../Link';

export type Props = {
  url: string;
  text: string;
};

/**
 * A link which automatically opens external links (URLs that aren't in its domain) in a new tab
 */
const AutoExternalLink: React.FC<Props> = ({ url, text }) => {
  const props = {
    url,
    text,
    isExternal: isExternalLink(url),
  };

  return <Link {...props} />;
};

export default AutoExternalLink;
