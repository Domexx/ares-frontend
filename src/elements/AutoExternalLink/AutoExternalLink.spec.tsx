import React from 'react';

import { render } from '@testing-library/react';

import AutoExternalLink from './AutoExternalLink';

describe('Auto External Link Component', () => {
  it('should render properly', () => {
    const { container } = render(<AutoExternalLink />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
