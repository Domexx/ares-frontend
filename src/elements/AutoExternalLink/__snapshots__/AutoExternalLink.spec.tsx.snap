// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auto External Link Component should render properly 1`] = `null`;
